test/unit/util_test.js

```diff
@@ -887,4 +890,94 @@ describe('Util', function () {
       });
     });
   });
+
+  if (os.platform() !== 'win32') {
+    describe('Util.isFileWritableOnlyByUser()', function () {
+      let tempDir = null;
+      let oldMask = null;
+
+      before(async function () {
+        tempDir = await fsPromises.mkdtemp(path.join(os.tmpdir(), 'permission_tests'));
+        oldMask = process.umask(0o000);
```

Comment:
does it read or modify the umask? didn't we provide explictly permissions when creating the directory?

Reply:
umask is respected by the mkdir, which means that even if we specify permissions 0777 it will be trimmed to 0755 by default, as the default umask is 022.

For the tests to proceed fine, the code sets the new umask to 000, but stores the original one in the variable `oldMask`. Then, after the tests it sets the umask back to it's original state.